### PR TITLE
Check workflow version is valid before publishing

### LINKED ISSUE #6670: I check if a workflow is valid before activation
- check if trigger has a next action
- check if trigger type is valid
- check if actions are valid

### PR DESCRIPTION
**Pull request by thomtrp**
Fix https://github.com/twentyhq/twenty/issues/6670

### CODE REVIEW

File: packages/twenty-server/src/modules/workflow/workflow-trigger/utils/assert-workflow-version-is-valid.ts [hunk 1]
```diff
@@ -0,0 +1,66 @@
...
+} from 'src/modules/workflow/workflow-trigger/workflow-trigger.exception';
+
+export function assertWorkflowVersionIsValid(
+  workflowVersion: Omit<WorkflowVersionWorkspaceEntity, 'trigger'> & {
+    trigger: WorkflowTrigger;
+  },
+) {
+  if (!workflowVersion.trigger) {
+    throw new WorkflowTriggerException(
+      'Workflow version does not contains trigger',
```
Comment (martmull) [inline packages/twenty-server/src/modules/workflow/workflow-trigger/utils/assert-workflow-version-is-valid.ts:18]:
```suggestion
      'Workflow version does not contain trigger',
```

File: packages/twenty-server/src/modules/workflow/common/workflow-common.workspace-service.ts [hunk 2]
```diff
@@ -35,7 +35,7 @@ export class WorkflowCommonWorkspaceService {
       );
     }
 
-    if (!workflowVersion.trigger || !workflowVersion.trigger?.type) {
+    if (!workflowVersion.trigger) {
```
Comment (martmull) [inline packages/twenty-server/src/modules/workflow/common/workflow-common.workspace-service.ts:38]: why not using your `assertWorkflowVersionIsValid` utils here?

Comment (martmull) [reply packages/twenty-server/src/modules/workflow/common/workflow-common.workspace-service.ts:38]: if you only need to check the trigger, maybe you can update `assertWorkflowVersionIsValid` to return after the trigger check

Comment (thomtrp) [reply packages/twenty-server/src/modules/workflow/common/workflow-common.workspace-service.ts:38]: I don't mind if the user stores a version with almost nothing in it. Maybe we'll even have an autosave for this so he does not lose its work. So the trigger could be totally empty, that's fine. The only important thing in my opinion is to validate before publishing the version. Does it make sense @martmull?

**Review by martmull — APPROVED** [review]

**Pull request merged**